consent/models.py

```diff
@@ -3,4 +3,153 @@
 Models for edX Enterprise's Consent application.
 """
 
-from __future__ import unicode_literals
+from __future__ import absolute_import, unicode_literals
+
+from consent.mixins import ConsentModelMixin
+from simple_history.models import HistoricalRecords
+
+from django.core.exceptions import ValidationError
+from django.db import IntegrityError, models
+from django.utils.translation import ugettext_lazy as _
+
+from model_utils.models import TimeStampedModel
+
+from enterprise.models import EnterpriseCustomer
+
+
+class DataSharingConsentQuerySet(models.query.QuerySet):
+    """
+    Customized QuerySets for the ``DataSharingConsent`` model.
+
+    When searching for any ``DataSharingConsent`` object, if it doesn't exist, return a single
+    ``ProxyDataSharingConsent`` object which behaves just like a ``DataSharingConsent`` object
+    except is not saved in the database until committed.
+    """
+
+    def get(self, *args, **kwargs):
+        """
+        Perform the query and returns a single object matching the given keyword arguments.
+
+        This customizes the queryset to return an instance of ``ProxyDataSharingConsent`` when
+        the searched-for ``DataSharingConsent`` instance does not exist.
+        """
+        try:
+            return super(DataSharingConsentQuerySet, self).get(*args, **kwargs)
+        except DataSharingConsent.DoesNotExist:
+            return ProxyDataSharingConsent(**kwargs)
```

Comment:
That's something I've already run into as part of the refactor, in fact, yeah. Have to think of a way around it.